lib/ansiblelint/__main__.py

```diff
@@ -114,9 +115,18 @@ def main() -> int:
                         options.verbosity, checked_files)
         matches.extend(runner.run())
 
-    for match in sorted(set(matches)):
+    # Assure we do not print duplicates and the order is consistent
+    matches = sorted(set(matches))
+
+    for match in matches:
         print(formatter.format(match, options.colored))
 
+    # If run under GitHub Actions we also want to emit output recognized by it.
+    if os.getenv('GITHUB_ACTIONS') == 'true' and os.getenv('GITHUB_WORKFLOW'):
+        formatter = formatters.AnnotationsFormatter(cwd, True)
+        for match in matches:
+            print(formatter.format(match))
```

Comment:
I just realized that there's no mechanism to disable this. I think it's important to have a CLI flag that would prevent annotations from being produced.

Reply:
There is no req to implement flags for this. If someone wants to avoid it, they can easily do `export GITHUB_WORKFLOW=` or `unset GITHUB_WORKFLOW` before calling it. Lets avoid adding feature nobody will ever need.

Reply:
How is relying on undocumented way of using implementation details is a good thing?

Reply:
I think that section added to README.rst is more than enough to document the behavior and it does not need to tell user how to avoid passing environment variables. It is outside our scope to teach user about how shell works, or how shell evaluates variables.